llvm/lib/ProfileData/InstrProfReader.cpp

```diff
@@ -676,8 +677,12 @@ Error RawInstrProfReader<IntPtrT>::readRawCounts(
       // A value of zero signifies the block is covered.
       Record.Counts.push_back(*Ptr == 0 ? 1 : 0);
     } else {
-      const auto *CounterValue = reinterpret_cast<const uint64_t *>(Ptr);
-      Record.Counts.push_back(swap(*CounterValue));
+      uint64_t CounterValue = swap(*reinterpret_cast<const uint64_t *>(Ptr));
+      if (CounterValue > MaxCounterValue)
+        WithColor::warning() << "counter value " + Twine(CounterValue) +
```

Comment:
I'd say something about the value being large, like "excessively large counter value ### suggests corrupted profile data"